src/devices/machine/ncr53c90.cpp

```diff
@@ -628,20 +647,15 @@ void ncr53c90_device::step(bool timeout)
 		break;
 
 	case INIT_XFR_FUNCTION_COMPLETE:
-		// wait for dma transfer to complete or fifo to drain
-		if ((dma_dir == DMA_OUT) && dma_command && !(status & S_TC0) && fifo_pos)
+		// wait for dma transfer to complete and fifo to drain
+		if (dma_command && (!(status & S_TC0) || fifo_pos))
```

Comment:
I'm working off the ncr53c90a/b datasheet for this one but it specifies that both of these conditions should be true to be accepted as a complete transfer.
![image](https://user-images.githubusercontent.com/63495610/216808396-69de422c-4232-41d1-8401-fe532b295a3b.png)